### DISCUSSION TITLE
Welcome to Civ2-clone Discussions!

### DISCUSSION THREAD

**Discussion by axx0** [axx0/Civ2-clone#29] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by LuKo555066** [comment]
Hello! I love the idea behind this project but at the moment it is a bit overwhelming - if I have any quick ideas I will post in issues section but a more detailed information what is already done and what still needs to be done would be very welcome.

**Reply by axx0** [reply]
Hi. I started this project years ago when I was learning c# with winforms. Then at the suggestion of civfanatics community I made this repo public and moved to Eto forms so it would run also on Linux/Mac. I'm not a pro programmer so the code is what it is (another guy has thankfully been cleaning up my mess).

Suggestions are welcome. I can spam issues section with stuff that needs to be done or populate the wiki with some sort of to-do list. Don't really know what's the best approach.

**Reply by LuKo555066** [reply]
It's important to make it work :) I can more or less read the code (despite my very low skill) so it is not that bad. I don't really have experience with github so if I write some code I will post it in issues section to be copied (as it is the case with production in cities) until I figure how to upload changes via github. Hope it is alright.

With regards the wiki or issues section - either is fine. As long as there is one place where everything is available I am happy :)

**Reply by axx0** [reply]
First make a fork of the repo, then make a new branch and commit your changes in the code there. Then you submit a PR on this repo so others can review the changes and if everything's ok I merge the changes into my repo. Use github for desktop for this. I use gitkraken. 

I'll see what I can do regarding wiki stuff.

**Comment by LuKo555066** [comment]
The main strength of this project is how easy it will be to mod the game. Do you plan to support it in any way or you just want to create a copy of the vanilla game?

**Reply by axx0** [reply]
I have no plans with mods.

**Reply by reubene** [reply]
I think mod support is important, although you want the default experience to be the classic civ2, it's more appealing to more people if it can be moded

**Reply by axx0** [reply]
It was poorly expressed on my side, of course the plan is to go beyond vanilla as much as possible and enable modding etc. I just had no clear vision how "modding tools" would look like. Of course Lua in combination with more open manipulation of game assets through various files is probably the way to go.

Also the project will have to eventually be taken over by somebody more versed in github. It's gone beyond my hobby knowledge of stuff and just there's no point just merging pull requests like crazy and screwing things up every now and then. So reubene if you're up for it please let me know.

**Reply by reubene** [reply]
You're able to add me as a collaborator under Settings -> Access -> Collaborators I think that allows me to merge PR's etc... I'm not a big github user either.

I don't really have much time to dedicate to this but I'm happy to help where I can. 

I'm hoping that we can get to a point where it works well enough for less technical people to contribute.

**Comment by reubene** [comment]
I'm adding support for Lua scripting, my plan is to implement the building and wonder effects in a Lua script that scenarios can override if they want this is an area of mod ability that is not possible in Totip. I'm planntto support all the Lua functions that Totip uses so their scripts will just work

**Reply by axx0** [reply]
That's brilliant.
